### PR TITLE
Add nitclk system tests

### LINKED ISSUE #1195: Proper TClk testing
We need to write system tests for `nitclk`. Fortunately NI recently made it possible to use NI-TClk with simulated drivers.

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] ~~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~
- [X] I've added tests applicable for this pull request

### What does this Pull Request accomplish?
* Adds nitclk system tests
    * Uses existing persistent niscope devices
### List issues fixed by this Pull Request below, if any.
* Fix #1195 

### What testing has been done?
* System

### CODE REVIEW

File: tools/system_test_jobs.groovy [hunk 1]
```diff
@@ -62,21 +62,33 @@ def genJob(driver, platform) {
...
+rem echo ghprbActualCommitAuthorEmail = %ghprbActualCommitAuthorEmail%
+rem echo ghprbPullDescription =         %ghprbPullDescription%
+rem echo ghprbPullId =                  %ghprbPullId%
+rem echo ghprbPullLink =                %ghprbPullLink%
+rem echo ghprbPullTitle =               %ghprbPullTitle%
+rem echo ghprbSourceBranch =            %ghprbSourceBranch%
+rem echo ghprbTargetBranch =            %ghprbTargetBranch%
+rem echo ghprbCommentBody =             %ghprbCommentBody%
+rem echo sha1 =                         %sha1%
+rem echo .
```
Comment (marcoskirsch) [inline tools/system_test_jobs.groovy]: get rid of commented out things?

**Comment by marcoskirsch** [comment]
Please confirm: NI-TClk must doesn't work at all with session-based simulation, hence we are required to use persistent simulation. If true, it should be documented.

File: build/templates/tox-system_tests.ini.mako [hunk 2]
```diff
@@ -4,46 +4,54 @@
...
     module_name = config['module_name']
     driver_name = config['driver_name']
-    if config['supports_nitclk']:
-        nitclk_env = 'py38-{}-nitclk_wheel,'.format(module_name)
+    if config['supports_nitclk'] or module_name == 'nitclk':
+        wheel_env_no_py = '{}-wheel_dep,'.format(module_name)
+        wheel_env = 'py38-' + wheel_env_no_py
+        uses_other_wheel = True
+        if module_name == 'nitclk':
+            other_wheel = 'niscope'
```
Comment (marcoskirsch) [inline build/templates/tox-system_tests.ini.mako]: comment

`# nitclk tests use the niscope module`

File: generated/nitclk/tox-system_tests.ini [hunk 3]
```diff
@@ -3,25 +3,29 @@
...
-envlist = py{35,36,37,38,py3}-nitclk-system_tests, py38-nitclk-coverage
+envlist = py38-nitclk-wheel_dep,py{35,36,37,38,py3}-nitclk-system_tests, py38-nitclk-coverage
 skip_missing_interpreters=True
 ignore_basepython_conflict=True
 # We put the .tox directory outside of the Jenkins workspace so that it isn't wiped with the rest of the repo
 toxworkdir = ../../../.tox
 
 [testenv]
 description =
+    nitclk-wheel_dep,: Build the niscope wheel
```
Comment (marcoskirsch) [inline generated/nitclk/tox-system_tests.ini]: add .. "because we use it in nitclk tests"

File: src/nifgen/system_tests/test_system_nifgen.py [hunk 4]
```diff
@@ -23,6 +23,15 @@ def session():
         yield simulated_session
 
 
+@pytest.fixture(scope='function')
+def session_5421():
+    with daqmx_sim_db_lock:
```
Comment (marcoskirsch) [inline src/nifgen/system_tests/test_system_nifgen.py]: The comment above "# We need a lock file so multiple tests aren't hitting the db at the same time" needs to be expanded and needs to reference the internal NI bug.

We should always reference a bug when we are working around a bug!

Specifically, it is not obvious why we grab a lock around init/close other than to a handful of NI driver engineers.

Comment (marcoskirsch) [reply src/nifgen/system_tests/test_system_nifgen.py]: I also think this should be own PR, it's not related to adding nitclk tests.

File: src/nitclk/system_tests/tests_system_nitclk.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+import datetime
+import fasteners
+import niscope
+import nitclk
+import os
+import pytest
+import tempfile
+
+
+# nitclk system tests require a different HW that the PXIe-5164 that we simulate else where. It does not support NI-TClk
```
Comment (marcoskirsch) [inline src/nitclk/system_tests/tests_system_nitclk.py]: I don't think this is accurate. PXIe-5164 does support NI-TClk.
What none of the drivers support is using NI-TClk combined with session-based simulation.

The persistent simulated devices don't need to be DAQmx-framework based devices.

**Comment by texasaggie97** [comment]
Switched to session based 5164 simulation. Now I am not sure what I did before that didn't work.

File: src/nitclk/system_tests/tests_system_nitclk.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+
+@pytest.fixture(scope='function')
+def session_5164():
+    with niscope.Session('', False, False, 'Simulate=1, DriverSetup=Model:5164;BoardType:PXIe') as simulated_session:
+        yield simulated_session
+
+
+@pytest.fixture(scope='function')
+def session_multiple_sessions():
```
Comment (marcoskirsch) [inline src/nitclk/system_tests/tests_system_nitclk.py]: Consider using consistent names to the fixtures that don't reference the specific model.
Like `single_niscope_session` and `multiple_niscope_sessions`

Comment (texasaggie97) [reply src/nitclk/system_tests/tests_system_nitclk.py]: Changed

**Pull request merged**